assets/src/edit-story/components/header/buttons.js

```diff
@@ -55,19 +59,94 @@ function PreviewButton() {
       meta: { isSaving },
       story: { link },
     },
+    actions: { saveStory },
   } = useStory();
 
+  const [previewLinkToOpenViaDialog, setPreviewLinkToOpenViaDialog] = useState(
+    null
+  );
+
   /**
    * Open a preview of the story in current window.
    */
   const openPreviewLink = () => {
     const previewLink = addQueryArgs(link, { preview: 'true' });
-    window.open(previewLink, 'story-preview');
+
+    // Start a about:blank popup with waiting message until we complete
+    // the saving operation. That way we will not bust the popup timeout.
+    let popup;
+    try {
+      popup = window.open('about:blank', PREVIEW_TARGET);
+      if (popup) {
+        popup.document.write('<!DOCTYPE html><html><body>');
+        // Output "waiting" message.
+        popup.document.write(
+          escapeHTML(
+            __('Please wait. Generating the preview...', 'web-stories')
+          )
+        );
+        // Force redirect to the preview URL after 5 seconds. The saving tab
```

Comment:
Where's the `5` coming from?

Reply:
No real reason, but there has to be some constraint. The backgrounded tab that started popup can be offloaded/paused/throttled.